### ISSUE TITLE
Features and documentation

### ISSUE THREAD

**Issue by Shreyans13** [Xayne/ts-free-monad#1]
What kind of features and optimisations (if any) we can expect in this repo.
Is this similar to free monard found in purescript ?

[Free Monard in puresctipt](https://pursuit.purescript.org/packages/purescript-free/7.0.0 )

#https://github.com/purescript/purescript-free

Or is this similar to something in Haskell. Also when can i expect this repo to have a proper documentation because this is much needed things in day-together projects. Also it helps in getting new features and PRs if you want to make this as a open source project

**Comment by Xayne** [comment]
Yep, this one is similar to this one: [Free Monad in purescript](https://pursuit.purescript.org/packages/purescript-free/7.0.0) 

There are no upcoming features or optimisations in my plans except the minor code clean-ups, but I am definitely open for proposals. 
 
Regarding documentation: I will do it in two weeks, likely at this weekend. 
I will appreciate if you are able to validate the documentation for sufficiency and clarity

**Comment by Xayne** [comment]
Sorry, looks like I can't do it in time I promised, currently I'am busy with other project.
@Shreyans13 let me know If you are really interested in it (or may be anyone else)
then I can change my priorities
